src/components/TodoList/TodoList.tsx

```diff
@@ -1,9 +1,33 @@
 import React from 'react';
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+
+type Props = {
+  todos: Todo[]
+  selectedFilter: string
+};
+
+export const TodoList: React.FC<Props> = ({ todos, selectedFilter }) => {
+  const visibleTodos = todos.filter(todo => {
+    switch (selectedFilter) {
+      case Filter.COMPLETED:
+        return todo.completed;
+      case Filter.ACTIVE:
+        return !todo.completed;
+      case Filter.ALL:
+      default:
+        return todo;
+    }
+  });
 
-export const TodoList: React.FC = () => {
   return (
     <section className="todoapp__main" data-cy="TodoList">
-      <div data-cy="Todo" className="todo completed">
+      {visibleTodos.map(todo => (
+        <TodoItem todo={todo} key={todo.id} />
+      ))}
+
+      {/* <div data-cy="Todo" className="todo completed">
```

Comment:
you can remove it

Reply:
i dont want till the end to not miss any thing in markup) thanks